hs_core/models.py

```diff
@@ -77,46 +81,58 @@ def validate_user_url(value):
 
 
 class ResourcePermissionsMixin(Ownable):
+    """Mix in can_* permission helper functions between users and resources."""
+
     creator = models.ForeignKey(User,
                                 related_name='creator_of_%(app_label)s_%(class)s',
                                 help_text='This is the person who first uploaded the resource',
                                 )
 
     class Meta:
+        """Define meta properties for ResourcePermissionsMixin, make abstract."""
+
         abstract = True
 
     @property
     def permissions_store(self):
+        """Use PERMISSIONS_DB constant. Unsure what 's' is here."""
         return s.PERMISSIONS_DB
 
     def can_add(self, request):
+        """Pass through can_change to determine if user can_add resources or resource files."""
```

Comment:
Any user can add resources in hydroshare. So may be better to document as: 

Pass through can_change to determine if user can make changes to a resource.